apps/site/pages/en/learn/modules/how-to-use-streams.md

```diff
@@ -0,0 +1,599 @@
+
+#### Key Methods and Events in Transform Streams
+
+Apart from all the methods and events in Duplex Streams there is
+
+- **[`_transform`][]**: This function is called internally to handle the flow of data between the readable and the writable parts. This MUST NOT be called by application code.
+
+#### Creating a Transform Stream
+
+To create a new transform stream we can pass an `options` object to the `Transform` constructor including a `transform` function which handles how the output data is computed from the input data using the `push` method.
+
+```cjs
+const { Transform } = require('node:stream');
+
+const upper = new Transform({
+  transform: function (data, enc, cb) {
+    this.push(data.toString().toUpperCase());
+    cb();
+  },
+});
+```
+
+This stream will take any input and output it in uppercase.
+
+## How to operate with streams
+
+When working with streams, we usually want to perform read from a source and write to a destination, possibly needing some transformation of the data in between. The following sections will cover different ways to do so.
+
+### .pipe()
+
+The [`.pipe()`][] method concatenates one readable stream to a writable (or transform) stream. Although this seems a simple way to achieve our goal, it delegates all error handling to the programmer making it difficult to get it right.
+
+The following example shows a pipe trying to output the current file in upper case to the console.
+
+```cjs
+const fs = require('node:fs');
+const { Transform } = require('node:stream');
+
+let errorCount = 0;
+const upper = new Transform({
+  transform: function (data, enc, cb) {
+    if (errorCount === 10) {
+      return cb(new Error('BOOM!'));
+    }
+    errorCount++;
+    this.push(data.toString().toUpperCase());
+    cb();
+  },
+});
+
+const readStream = fs.createReadStream(__filename, { highWaterMark: 1 });
+const writeStream = process.stdout;
+
+readStream.pipe(upper).pipe(writeStream);
+
+readStream.on('close', () => {
+  console.log('Readable stream closed');
+});
+
+upper.on('close', () => {
+  console.log('Transform stream closed');
+});
+
+upper.on('error', err => {
+  console.error('\nError in transform stream:', err.message);
+});
+
+writeStream.on('close', () => {
+  console.log('Writable stream closed');
+});
+```
+
+After writing 10 characters `upper` will return an error in the callback, which will cause the stream to close. However, the other streams won’t be notified resulting in memory leaks. The output will be:
+
+```bash
+CONST FS =
+Error in transform stream: BOOM!
+Transform stream closed
+```
+
+### pipeline()
+
+To avoid the pitfalls low level complexity of the `.pipe()` method, in most cases it is recommended to use [`pipeline()`][] method. This method is a safer and more robust way to pipe streams together, handling errors and cleanup automatically.
+
+The following example demonstrates how using`pipeline()`prevents the pitfalls of the previous example:
```

Comment:
```suggestion
To avoid the pitfalls and low-level complexity of the `.pipe()` method, in most cases, it is recommended to use the [`pipeline()`][] method. This method is a safer and more robust way to pipe streams together, handling errors and cleanup automatically.

The following example demonstrates how using `pipeline()` prevents the pitfalls of the previous example:
```